geomstats/geometry/hyperbolic_space.py

```diff
@@ -265,45 +266,51 @@ def exp(self, tangent_vec, base_point):
-        coef_2 += mask_0_float * (
-                  1. + SINH_TAYLOR_COEFFS[3] * norm_tangent_vec ** 2
-                  + SINH_TAYLOR_COEFFS[5] * norm_tangent_vec ** 4
-                  + SINH_TAYLOR_COEFFS[7] * norm_tangent_vec ** 6
-                  + SINH_TAYLOR_COEFFS[9] * norm_tangent_vec ** 8)
-        # This avoids dividing by 0.
-        norm_tangent_vec += mask_0_float * 1.0
-        coef_1 += mask_else_float * (gs.cosh(norm_tangent_vec))
-        coef_2 += mask_else_float * (
-            (gs.sinh(norm_tangent_vec) / (norm_tangent_vec)))
-
-        exp = (gs.einsum('ni,nj->nj', coef_1, base_point)
-               + gs.einsum('ni,nj->nj', coef_2, tangent_vec))
-
-        hyperbolic_space = HyperbolicSpace(dimension=self.dimension)
-        exp = hyperbolic_space.regularize(exp)
-        return exp
+        if(self.point_type==None):
+            tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=2)
+            base_point = gs.to_ndarray(base_point, to_ndim=2)
+
+            sq_norm_tangent_vec = self.embedding_metric.squared_norm(
+                    tangent_vec)
+            norm_tangent_vec = gs.sqrt(sq_norm_tangent_vec)
+
+            mask_0 = gs.isclose(sq_norm_tangent_vec, 0.)
+            mask_0 = gs.to_ndarray(mask_0, to_ndim=1)
+            mask_else = ~mask_0
+            mask_else = gs.to_ndarray(mask_else, to_ndim=1)
+            mask_0_float = gs.cast(mask_0, gs.float32)
+            mask_else_float = gs.cast(mask_else, gs.float32)
+
+            coef_1 = gs.zeros_like(norm_tangent_vec)
+            coef_2 = gs.zeros_like(norm_tangent_vec)
+
+            coef_1 += mask_0_float * (
+                      1. + COSH_TAYLOR_COEFFS[2] * norm_tangent_vec ** 2
+                      + COSH_TAYLOR_COEFFS[4] * norm_tangent_vec ** 4
+                      + COSH_TAYLOR_COEFFS[6] * norm_tangent_vec ** 6
+                      + COSH_TAYLOR_COEFFS[8] * norm_tangent_vec ** 8)
+            coef_2 += mask_0_float * (
+                      1. + SINH_TAYLOR_COEFFS[3] * norm_tangent_vec ** 2
+                      + SINH_TAYLOR_COEFFS[5] * norm_tangent_vec ** 4
+                      + SINH_TAYLOR_COEFFS[7] * norm_tangent_vec ** 6
+                      + SINH_TAYLOR_COEFFS[9] * norm_tangent_vec ** 8)
+            # This avoids dividing by 0.
+            norm_tangent_vec += mask_0_float * 1.0
+            coef_1 += mask_else_float * (gs.cosh(norm_tangent_vec))
+            coef_2 += mask_else_float * (
+                (gs.sinh(norm_tangent_vec) / (norm_tangent_vec)))
+
+            exp = (gs.einsum('ni,nj->nj', coef_1, base_point)
+                   + gs.einsum('ni,nj->nj', coef_2, tangent_vec))
+
+            hyperbolic_space = HyperbolicSpace(dimension=self.dimension)
+            exp = hyperbolic_space.regularize(exp)
+            return exp
+
+        if(self.point_type=='Poincare'):
+            raise NotImplementedError(
+                'The computation of the poincare log'
+                ' is not implemented.')
```

Comment:
Maybe `The computation of the Exp in Poincare representation is not implemented` ? would be slightly more precise?

(also, it's Exp not Log, I think you didn't change it after copy-pasting)